File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.util.*;
public class Main{
    static int N;
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        int[] arr = new int[N];
        for (int i = 0; i < N; i++) {
            arr[i]=sc.nextInt();
        }
        Arrays.sort(arr);
        int sum = sc.nextInt(); // 찾고자하는 부분합
        int cnt=0; // 정답
        int start=0;
        int end=N-1;

        while (start<end) {
            if (arr[start]+arr[end]==sum) {
                cnt++;
                start++;
                end--;
            } else if (arr[start]+arr[end]<sum) {
                start++;
            } else end--;
        }
        System.out.println(cnt);
    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');

let n = Number(input[0]);
let arr = input[1].split(" ").map(Number);
let x = Number(input[2]);
arr.sort((a, b) => a - b);
let cnt = 0;

let start=0;
let end=n-1;

while(start<end) {
    let sum=arr[start]+arr[end];
    if(sum==x) {
        cnt++;
        start++;
        end--;
    } else if(sum>x) {
        end--; // 현재 합이 x보다 크다면, 합을 줄이기 위해 end를 감소
    } else start++;
}
console.log(cnt);